### PR TITLE
feat: use live regions to announce travel aid updates

### PR DESCRIPTION
**Pull request by rosvik**
Refactors travel aid to use [live regions](https://reactnative.dev/docs/accessibility#accessibilityliveregion-android) to announce changes on the page. Since iOS doesn't support live regions, the `useLiveRegionAnnouncement` hook is added, that triggers screen reader announcements on iOS when a11yLabel changes. There's also a wrapper component `LiveRegionWrapper` that can be used for non-interactive components in JSX.

### Example usage of live regions

There are two main ways to implement live regions

```tsx
const a11yProps = useLiveRegionAnnouncement(a11yLabelThatChangesOverTime, true);
return (
	<InteractiveComponentWithAnnoucements
		accessibilityRole="button"
		onPress={() => {}}
		{...a11yProps}
	/>
)
```

or

```tsx
return (
	<LiveRegionWrapper accessibilityLabel={a11yLabelThatChangesOverTime}>
		<NonInteractiveComponent />
	</LiveRegionWrapper>
)
```

closes https://github.com/AtB-AS/kundevendt/issues/19525

### CODE REVIEW

File: src/components/screen-reader-announcement/use-live-region-announcement.ts [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+  accessible: true;
+  accessibilityLabel?: string;
+  accessibilityLiveRegion?: 'polite';
+};
+
+/**
+ * When a11yLabel changes, announce it to screen readers without interrupting
+ * ongoing speach. Returns accessibility props that should be added to the
+ * relevant component. For non-interactive components, consider using
+ * `AnnouncementWrapper` instead.
```
Comment (gorandalum) [inline src/components/screen-reader-announcement/use-live-region-announcement.ts:14]:
```suggestion
 * `LiveRegionWrapper` instead.
```

File: src/components/screen-reader-announcement/use-live-region-announcement.ts [hunk 2]
```diff
@@ -0,0 +1,37 @@
+import {useEffect} from 'react';
+import {AccessibilityInfo, Platform} from 'react-native';
+
+type LiveRegionAccessibilityProps = {
+  accessible: true;
+  accessibilityLabel?: string;
+  accessibilityLiveRegion?: 'polite';
+};
```
Comment (gorandalum) [inline src/components/screen-reader-announcement/use-live-region-announcement.ts:8]: Maybe deduce this type from `AccessibilityProps`?

File: src/travel-aid/TravelAidScreenComponent.tsx [hunk 3]
```diff
@@ -228,28 +205,36 @@ const TravelAidSection = ({
...
-            accessibilityLabel={focussedStateA11yLabel}
-            style={styles.stopHeaderContainer}
-          >
-            <View>
+          <View accessible style={styles.stopHeaderContainer}>
+            <LiveRegionWrapper
+              accessibilityLabel={
+                getStopHeader(status, t) + screenReaderPause + quayName
+              }
+            >
```
Comment (gorandalum) [inline src/travel-aid/TravelAidScreenComponent.tsx:232]: Should there be two layers of `accessible` here?

**Review by gorandalum — APPROVED** [review]
This looks great!

**Review by jorelosorio — APPROVED** [review]
We had a talk while exploring this option, but I haven't tested with the cases I covered in my solution to handle multiple messages and priorities. The code looks good based in our discussions.

File: src/components/screen-reader-announcement/use-live-region-announcement.ts [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+      AccessibilityInfo.announceForAccessibilityWithOptions(a11yLabel, {
+        queue: true,
+      });
+    }
+  }, [a11yLabel, enabled]);
+
+  return {
+    accessible: true,
+    accessibilityLabel: a11yLabel,
+    accessibilityLiveRegion: enabled ? 'polite' : undefined,
```
[the remark this answers is not in the document]
Comment (rosvik) [reply src/components/screen-reader-announcement/use-live-region-announcement.ts:35]: I don't think keeping the android specific props is an issue for iOS here. I guess it wouldn't hurt to omit it either, but it's extra code with no effect.

File: src/travel-aid/TravelAidScreenComponent.tsx (hunk 3 above)
Comment (rosvik) [reply src/travel-aid/TravelAidScreenComponent.tsx:232]: The topmost `accessible` here is to better match the sketches, where the time and stop place is one focus area. It is, however, handled differently on iOS and Android. Talkback makes the two children focusable, while VoiceOver focuses the parent. Both platforms have the same announcements. I've clarified with @hannahATB that it's ok to have it this way.

**Pull request merged**